src/lm_buddy/jobs/configs/base.py

```diff
@@ -38,3 +51,12 @@ def to_tempfile(self, *, name: str = "config.yaml", dir: str | Path | None = Non
             config_path = Path(tmpdir) / name
             self.to_yaml_file(config_path)
             yield config_path
+
+    @abstractmethod
```

Comment:
I was a bit unhappy with this method, but essentially I need a way of extracting all the _input_ `AssetPath`s off a `LMBuddyJobConfig`. This is so we can link them as input artifacts in the lineage generation.

The fields are not standard across jobs, so I left it abstract and forced every job config to implement it individually.